rotkehlchen/db/updates.py

```diff
@@ -154,6 +155,53 @@ def update_rpc_nodes(self, to_version: int) -> None:
             new_default_nodes=new_default_nodes,
         )
 
+    def update_contracts(self, to_version: int) -> None:
+        """Updates evm contracts and ABIs in globaldb"""
+        updated_contracts: dict
+        success, updated_contracts = self._get_file_content(  # type: ignore[assignment]  # updated_contracts is a dictionary for sure  # noqa: E501
+            file_url=CONTRACTS_URL.format(branch=self.branch, version=to_version),
+            update_type=UpdateType.CONTRACTS,
+        )
+        if success is False:
+            return
+
+        log.info(f'Applying update for contracts to v{to_version}')
+        new_abi_data = []
+        for single_abi_data in updated_contracts['abis_data']:
+            new_abi_data.append((
+                single_abi_data['id'],
+                single_abi_data['value'],
+                single_abi_data['name'],
+            ))
+
+        new_contracts_data = []
+        for single_contract_data in updated_contracts['contracts_data']:
+            new_contracts_data.append((
+                single_contract_data['address'],
+                single_contract_data['chain_id'],
+                single_contract_data.get('name'),
+                single_contract_data['abi'],
+                single_contract_data.get('deployed_block'),
+            ))
+
+        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+            write_cursor.execute('DELETE FROM contract_data')
+            write_cursor.execute('DELETE FROM contract_abi')
```

Comment:
why delete what we have? If you do incremental versions you won't delete anything otherwise what was provided in v1 will be lost in v2.

The good thing to go with versioning here is that you avoid downloading huge files

Reply:
Mmm yes it makes sense, however the problem is that contracts reference abi ids. And to make sure that the referenced abi id are the needed ones I think it will be needed to provide all abis and all contracts here.